### PR TITLE
[NEW UI] Update page - sentry user feedback

### PR DESCRIPTION
**Pull request by ga-devfront**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | Implement user feedback modal after logs error on update page.
| Type?             | new feature
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | N/A
| Sponsor company   | @PrestaShopCorp
| How to test?      | Try to update with error (you can manually add error on any task). Then the update stop, you can click on send error button. If you submit the form, you retrieve well the error on sentry.

### CODE REVIEW

File: classes/Traits/displayErrorReportDialogTrait.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\Traits;
+
+use PrestaShop\Module\AutoUpgrade\AjaxResponseBuilder;
+use PrestaShop\Module\AutoUpgrade\DocumentationLinks;
+use PrestaShop\Module\AutoUpgrade\Twig\PageSelectors;
+use Symfony\Component\HttpFoundation\JsonResponse;
+
+trait displayErrorReportDialogTrait
```
Comment (M0rgan01) [inline classes/Traits/displayErrorReportDialogTrait.php]:
```suggestion
trait DisplayErrorReportDialogTrait
```

File: classes/Traits/displayErrorReportDialogTrait.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+{
+    public function submitErrorReport(): JsonResponse
+    {
+        return AjaxResponseBuilder::hydrationResponse(
+            PageSelectors::DIALOG_PARENT_ID,
+            $this->getTwig()->render(
+                '@ModuleAutoUpgrade/dialogs/dialog-error-report.html.twig',
+                [
+                    'dialogSize' => 'sm',
+                    'title' => $this->upgradeContainer->getTranslator()->trans('Send error report?'),
```
Comment (M0rgan01) [inline classes/Traits/displayErrorReportDialogTrait.php]:
```suggestion
                    'title' => $this->upgradeContainer->getTranslator()->trans('Send error report ?'),
```

Comment (Quetzacoalt91) [reply classes/Traits/displayErrorReportDialogTrait.php]: Actually, there is no space before a question mark in English :grimacing:

File: classes/Traits/displayErrorReportDialogTrait.php [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+{
+    public function submitErrorReport(): JsonResponse
+    {
+        return AjaxResponseBuilder::hydrationResponse(
+            PageSelectors::DIALOG_PARENT_ID,
+            $this->getTwig()->render(
+                '@ModuleAutoUpgrade/dialogs/dialog-error-report.html.twig',
+                [
+                    'dialogSize' => 'sm',
+                    'title' => $this->upgradeContainer->getTranslator()->trans('Send error report ?'),
```
Comment (Hlavtox) [inline classes/Traits/displayErrorReportDialogTrait.php]:
```suggestion
                    'title' => $this->upgradeContainer->getTranslator()->trans('Send error report?'),
```

**Review by Quetzacoalt91 — APPROVED** [review]

**Review by Quetzacoalt91 — CHANGES_REQUESTED** [review]
## PSR4 issue

```
Class PrestaShop\Module\AutoUpgrade\Traits\DisplayErrorReportDialogTrait located in ./classes/Traits/displayErrorReportDialogTrait.php does not comply with psr-4 autoloading standard. Skipping.
```

Please rename the file `classes/Traits/displayErrorReportDialogTrait.php` with the proper case.

## Errors unexpectedly sent to Sentry

Two errors were sent to Sentry without any action from me, which was not the case before.
image

**Comment by ga-devfront** [comment]
> ## PSR4 issue
> ```
> Class PrestaShop\Module\AutoUpgrade\Traits\DisplayErrorReportDialogTrait located in ./classes/Traits/displayErrorReportDialogTrait.php does not comply with psr-4 autoloading standard. Skipping.
> ```
> 
> Please rename the file `classes/Traits/displayErrorReportDialogTrait.php` with the proper case.
> 
> ## Errors unexpectedly sent to Sentry
> Two errors were sent to Sentry without any action from me, which was not the case before. image

fixed

**Comment by tblivet** [comment]
@ga-devfront, I noticed that in the design, we seem to have a textarea instead of a text input for the description section.

image

File: _dev/src/ts/dialogs/SendErrorReportDialog.ts [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+    }
+
+    return form;
+  }
+
+  #onSubmit = (event: SubmitEvent) => {
+    event.preventDefault();
+
+    const logsViewer = document.querySelector(
+      '[data-component="progress-tracker"] [data-component="logs-viewer"]'
```
Comment (tblivet) [inline _dev/src/ts/dialogs/SendErrorReportDialog.ts]: Do we really need both `[data-component="progress-tracker"] [data-component="logs-viewer"]`? Only `[data-component="logs-viewer"]` should be sufficient.

File: _dev/src/ts/dialogs/SendErrorReportDialog.ts [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    );
+
+    const logs: Logs = {};
+
+    const logsContent = logsViewer?.querySelector(
+      '[data-slot-component="scroll"] [data-slot-component="list"]'
```
Comment (tblivet) [inline _dev/src/ts/dialogs/SendErrorReportDialog.ts]: Do we really need both `[data-slot-component="scroll"] [data-slot-component="list"]`? Only `[data-slot-component="list"]` should be sufficient.

File: _dev/src/ts/routing/ScriptHandler.ts [hunk 6]
```diff
@@ -42,7 +46,7 @@ export default class ScriptHandler {
 
   /**
    * @public
-   * @param {string} routeName - The name of the route to load his associated script.
+   * @param {string} scriptID - The name of the route to load his associated script.
```
Comment (tblivet) [inline _dev/src/ts/routing/ScriptHandler.ts]:
```suggestion
   * @param {string} scriptID - The ID of the route to load his associated script.
```

File: views/templates/dialogs/dialog-error-report.html.twig [hunk 7]
```diff
@@ -6,54 +6,56 @@
...
   </div>
 {% endblock %}
 
 {% block dialog_extra_content_inner %}
-  <div class="dialog__error-report-form">
+  <form class="dialog__error-report-form" id="form-error-feedback" name="form-error-feedback">
     <div class="form-group">
       <label for="userEmail">
-        Email (optional)
+        {{ 'Email'|trans({}) }} {{ '(optional)'|trans({}) }}
```
Comment (tblivet) [inline views/templates/dialogs/dialog-error-report.html.twig]:
```suggestion
        {{ 'Email (optional)'|trans({}) }}
```

File: views/templates/dialogs/dialog-error-report.html.twig [hunk 8]
```diff
@@ -6,54 +6,56 @@
...
-        placeholder="Enter your email address"
+        name="email"
+        placeholder="{{ 'Enter your email address'|trans({}) }}"
       />
     </div>
 
     <div class="form-group">
       <label for="errorDescription">
-        Description (optional)
+        {{ 'Description'|trans({}) }} {{ '(optional)'|trans({}) }}
```
Comment (tblivet) [inline views/templates/dialogs/dialog-error-report.html.twig]:
```suggestion
        {{ 'Description (optional)'|trans({}) }}
```

**Review by tblivet — DISMISSED** [review]

File: classes/Traits/displayErrorReportDialogTrait.php (hunk 2 above)
Comment (M0rgan01) [reply classes/Traits/displayErrorReportDialogTrait.php]: 😫

File: _dev/src/ts/pages/UpdatePageUpdate.ts [hunk 9]
```diff
@@ -32,8 +33,9 @@ export default class UpdatePageUpdate extends UpdatePage {
   public beforeDestroy = () => {
     this.#progressTracker.beforeDestroy();
 
-    this.#restoreAlertForm?.removeEventListener('submit', this.#handleRestoreSubmit);
-    this.#restoreButtonForm?.removeEventListener('submit', this.#handleRestoreSubmit);
+    this.#restoreAlertForm?.removeEventListener('submit', this.#handleSubmit);
+    this.#restoreButtonForm?.removeEventListener('submit', this.#handleSubmit);
+    this.#submitErrorReportForm?.addEventListener('submit', this.#handleSubmit);
```
Comment (Quetzacoalt91) [inline _dev/src/ts/pages/UpdatePageUpdate.ts]: `addEventListener` ? Or did you mean `removeEventListener`

**Review by Quetzacoalt91 — APPROVED** [review]

**Pull request merged**